src/interval_sets.jl

```diff
@@ -469,16 +469,15 @@ function intersection_isless_fn(::TrackEachEndpoint)
 end
 
 """
-    find_intersections(
-        x::AbstractVector{<:AbstractInterval},
-        y::AbstractVector{<:AbstractInterval}
-    )
+    find_intersections(x, y)
 
 Returns a `Vector{Vector{Int}}` where the value at index `i` gives the indices to all
-intervals in `y` that intersect with `x[i]`.
+intervals in `y` that intersect with `x[i]`. Calls collect on the arguments if they
+aren't already arrays.
 """
-find_intersections(x, y) = find_intersections(vcat(x), vcat(y))
-function find_intersections(x::AbstractVector{<:AbstractInterval}, y::AbstractVector{<:AbstractInterval})
+find_intersections(x, y) = find_intersections_(collect(x), collect(y))
+function find_intersections_(x::AbstractVector, y::AbstractVector)
+    (isempty(x) || isempty(y)) && return Vector{Int}[]
```

Comment:
May as well avoid collecting types that are already `AbstractVector`s such as ranges:
```suggestion
find_intersections(x, y) = find_intersections(collect(x), collect(y))
function find_intersections(x::AbstractVector, y::AbstractVector)
    (isempty(x) || isempty(y)) && return Vector{Int}[]
```

Reply:
If `collect` is poorly implemented the proposed modification could lead to a stack overflow. I've added a new method, and kept `find_intersections_`